Add allow-not-found input argument

croemheld opened the pull request:
Closes #42

This PR adds the `allow-not-found` input argument to `download-artifact` that, when set to `true`, will not fail the action when an artifact with the specified `name` input argument is not found.

If the artifact is not found, the output of `download-path` of the action will contain an empty string instead of the would-be download path. This can be checked in the subsequent step via `${{ steps.<id>.outputs.download-path != '' }}` as can be seen in the updated workflow test.yml file. Additionally, a warning is printed via `core.warning` if the artifact is not found while `allow-not-found` is set to true.

If `allow-not-found` is set to false (default behavior), the action behaves as it did before this PR.

croemheld commented:
@robherley As of now, there still seem to be proponents of this feature. Is there a chance that this is still going to be included or are there other plans currently?

DaUs3103 reviewed (commented):
[CHANGELOG.md](https://github.com/actions/download-artifact/files/15302218/CHANGELOG.md)

DaUs3103 reviewed (approved):
[README.md](https://github.com/actions/download-artifact/files/15302248/README.md)

DaUs3103 commented:
[README.md](https://github.com/actions/download-artifact/files/15302251/README.md)

croemheld commented:
@DaUs3103 Thanks for reviewing! I wasn't sure how to interpret your comment, should I update the README.md and CHANGELOG.md files as well?

rrundle-dave commented:
@croemheld I think he's saying update the README because any new option that gets added needs to be documented so users know it's an available option.

croemheld commented:
@DaUs3103 @rrundle-dave Sorry for the long delay, I rebased on top of the latest changes and updated the README.md file with the new input argument as well.
